examples/remote-sensing/plot_clustering.py

```diff
@@ -0,0 +1,373 @@
+"""
+====================================================================
+Remote sensing: Clustering on hyperspectral/sar images with Riemannian
+geometry
+====================================================================
+
+This example compares clustering pipelines based on covariance matrices for
+hyperspectral/sar data custering.
+"""
+
+# Author: Ammar Mian
+
+import os
+import urllib.request
+import numpy as np
+import numpy.linalg as la
+from numpy.typing import ArrayLike
+import matplotlib.pyplot as plt
+
+from sklearn.pipeline import Pipeline
+from pyriemann.estimation import Covariances
+from pyriemann.clustering import Kmeans
+from scipy.io import loadmat
+from typing import Tuple, Dict
+
+from numpy.lib.stride_tricks import sliding_window_view
+from math import ceil 
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from numpy.typing import ArrayLike
+
+
+class RemoveMeanImage(BaseEstimator, TransformerMixin):
+    """ Remove the global mean of an three-dimensional image.
+    """
+    def __init__(self):
+        pass
+
+    def fit(self, X: ArrayLike, y = None):
+        return self
+
+    def transform(self, X: ArrayLike):
+        return X - np.mean(X, axis=(0, 1))
+
+    def fit_transform(self, X: ArrayLike, y = None):
+        return self.fit(X).transform(X)
+
+
+class PCAImage(BaseEstimator, TransformerMixin):
+    """PCA on an image of shape (n_rows, n_columns, n_features).
+
+    Parameters
+    ----------
+    n_components : int
+        Number of components to keep.
+    """
+
+    def __init__(self, n_components: int):
+        assert n_components > 0, 'Number of components must be positive.'
+        self.n_components = n_components
+
+    def fit(self, X: ArrayLike, y = None):
+        return self
+
+    def transform(self, X: ArrayLike):
+        if self.n_components == X.shape[2]:
+            return X
+        return pca_image(X, self.n_components)
+
+    def fit_transform(self, X: ArrayLike, y = None):
+        return self.fit(X).transform(X)
+
+
+class SlidingWindowVectorize(BaseEstimator, TransformerMixin):
+    """Slinding window over an image of shape (n_rows, n_columns, n_features).
+
+    Parameters
+    ----------
+    window_size : int
+        Size of the window.
+
+    overlap : int
+        Overlap between windows. Default is 0.
```

Comment:
In pyRiemann, we don't jump a line between parameters.
```suggestion
        Size of the window.
    overlap : int, default=0
        Overlap between windows.
```